app/helpers/katello/subscription_mailer_helper.rb

```diff
@@ -0,0 +1,17 @@
+module Katello
+  module SubscriptionMailerHelper
+    include Rails.application.routes.url_helpers
+
+    def report_url
+      params = { format: 'html', template_id: ReportTemplate.find_by(name: "Subscription - Entitlement Report").id , input_values: { TemplateInput.find_by_name("Days from Now").id.to_s  => { value: UserMailNotification.first.mail_query.to_s } } }
```

Comment:
`ReportTemplate.find_by(name: "Subscription - Entitlement Report").id` is used twice in this method; I suggest DRYing it up by assigning it to a variable.

Reply:
For readability, it might be better to also assign `TemplateInput.find_by_name("Days from Now").id.to_s` and the `mail_query` value to variables as well.